types/index.d.ts

```diff
@@ -25,10 +25,10 @@ declare namespace fastifyOauth2 {
     scope?: string[];
     credentials: Credentials;
     callbackUri: string;
-    callbackUriParams?: Object;
-    tokenRequestParams?: Object;
-    generateStateFunction?: Function;
-    checkStateFunction?: Function;
+    callbackUriParams?: Record<string, string>;
+    tokenRequestParams?: Record<string, string>;
+    generateStateFunction?: (request: FastifyRequest) => string;
+    checkStateFunction?: (request?: FastifyRequest, callback?: (err: any) => void) => string | void;
```

Comment:
Arguments is not optional and it does need to return any result.

```suggestion
    checkStateFunction?: (request: FastifyRequest, callback: (err: any) => void) => void;
```

Reply:
updated the code accordingly :)